js/main.js

```diff
@@ -30,14 +51,31 @@ function formSubmitted(event) {
     comment: $textInputs[2].value,
     entryId: data.nextEntryId
   };
-  data.nextEntryId++;
-  data.entries.push(newEntry);
+  // Editing or adding items
+  if (data.editing) {
+    data.entries[data.editing - 1].title = newEntry.title;
+    data.entries[data.editing - 1].photoUrl = newEntry.photoUrl;
+    data.entries[data.editing - 1].comment = newEntry.comment;
+    data.editing = null;
+  } else {
+    data.nextEntryId++;
+    data.entries.push(newEntry);
+  }
+  // Adding all items into the list from the data model
+  $journalFeedList.innerHTML = '';
+  for (var i = 0; i < data.entries.length; i++) {
+    var tempDomTree = creatingJournalEntry(data.entries[i]);
+    $journalFeedList.prepend(tempDomTree);
+    tempDomTree = {};
+  }
```

Comment:
Instead of recreating the entire list, you should just find the original DOM tree for the entry being edited and replace it with a new DOM tree with the edited data. This is a lot of work for the browser as it is written right now.